--- origsrc/glibmm-2.16.4/examples/compose/main.cc	2008-06-16 16:32:24.000000000 -0500
+++ src/glibmm-2.16.4/examples/compose/main.cc	2008-07-06 20:21:53.781250000 -0500
@@ -40,8 +40,13 @@
                         ustring::format(std::fixed, std::setprecision(1), a / b * 100.0))
     << std::endl
     << ustring::compose("a : b = [%1|%2]",
+#ifdef _GLIBCXX_USE_WCHAR_T
                         ustring::format(std::setfill(L'a'), std::setw(i), ""),
                         ustring::format(std::setfill(L'b'), std::setw(40 - i), ""))
+#else
+                        ustring::format(std::setfill('a'), std::setw(i), ""),
+                        ustring::format(std::setfill('b'), std::setw(40 - i), ""))
+#endif
     << std::endl;
 }
 
